### DISCUSSION TITLE
PostgREST schema cache reloading is not working

### DISCUSSION THREAD

**Discussion by dreinon** [supabase/supabase#3186] [Questions] [answered]
I'm getting the following error trying to query a foreign table through a foreign key in a table:
```python
db.table('alumnos_matriculas').select('mat_id, precios (*)').execute()[data]
```

Response:
```json
{
  "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
  "message": "Could not find a relationship between alumnos_matriculas and precios  in the schema cache"
}
```

I looked at the PostgREST docs and found [how to reload cache with notify](https://postgrest.org/en/v8.0/schema_cache.html#reloading-with-notify) and [how to do it automatically](https://postgrest.org/en/v8.0/schema_cache.html#automatic-schema-cache-reloading) but after trying both, none is working.

I also tried to restart the server, and tried to unapply and reapply the foreign key constraint in the column, but again, none works.

I can query other foreign tables through a foreign key in other tables, but this one is not working,

Any ideas?

**Comment by awalias** [comment]
hmm very strange, we do trigger a reload in postgrest automatically each time a table/column etc is added. If it doesn't work after a server restart then perhaps it is something to do with the relationship itself - can you share exactly how the entities are related

**Comment by dreinon** [comment]
Yes, the foreign table i'm querying has serial primary key, other unsignificant columns and a text column referencing the other table's primary key, which is text also

**Reply by dreinon** [reply]
So:

Table A:
a_id: serial
.
.
.
fk_b_id: text

Table B:
b_id: text
.
.
.

**Comment by steve-chavez** [comment]
> db.table('alumnos_matriculas').select('mat_id, precios (*)').execute()[data]

> {
>  "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
>  "message": "Could not find a relationship between profiles and assigned_departments in the schema cache"
> }

Hm, the error mentions `profiles` and `assigned_departments`, but the sample js snippet above doesn't show any of those tables. 

Try running/sharing this query result:

```sql
select  * from information_schema.key_column_usage where table_name in ('profiles', 'assigned_departments');
```

**Reply by dreinon** [reply]
Sorry, this is because I copied the error from other guy with the same error. It's edited now.

**Reply by dreinon** [reply]
I'll query and share though

**Reply by dreinon** [reply]
I ran the query and got the following:
```sql
-- Query
select  * from information_schema.key_column_usage where table_name in ('precios', 'alumnos_matriculas');
```
Results
| constraint_catalog | constraint_schema | constraint_name                    | table_catalog | table_schema | table_name         | column_name | ordinal_position | position_in_unique_constraint |
| ------------------ | ----------------- | ---------------------------------- | ------------- | ------------ | ------------------ | ----------- | ---------------- | ----------------------------- |
| postgres           | public            | precios_pkey                       | postgres      | public       | precios            | pre_id      | 1                |                               |
| postgres           | public            | alumnos_matriculas_mat_alu_id_fkey | postgres      | public       | alumnos_matriculas | mat_alu_id  | 1                | 1                             |
| postgres           | public            | alumnos_matriculas_pkey            | postgres      | public       | alumnos_matriculas | mat_id      | 1                |                               |
| postgres           | public            | alumnos_matriculas_mat_precio_fkey | postgres      | public       | alumnos_matriculas | mat_precio  | 1                | 1                             |

**Comment by dreinon** [comment, marked as answer]
Found the problem!
It's how python client is interpreting the query. I noticed table name `precios` had a space after it in the error message:
```json
{
  "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
  "message": "Could not find a relationship between alumnos_matriculas and precios  in the schema cache"
}
```

Then, I found out the query my code was the following:
```python
db.table('alumnos_matriculas') .select('mat_id, precios (*)').execute()
```

And when I removed the space between table_name `precios` and `(*)` got it working:
```python
db.table('alumnos_matriculas') .select('mat_id, precios(*)').execute()
```
